src/AppBundle/View/I18nResponseBuilder.php

```diff
@@ -0,0 +1,169 @@
+ *
+ * [
+ *   'de' => ['German translation'],
+ *   'en' => ['English translation'],
+ * ]
+ *
+ * Furthermore it is possible to sort all the violations by the property which caused the violation:
+ *
+ * [
+ *   'property' => [
+ *     [
+ *       'de' => ['German translation'],
+ *       'en' => ['English translation'],
+ *     ]
+ *   ],
+ *   'another_property' => [
+ *     [
+ *       // ...
+ *     ]
+ *   ]
+ * ]
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ */
+class I18nResponseBuilder implements I18nResponseBuilderInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * Constructor.
+     *
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildTranslatableResponse(
+        ConstraintViolationListInterface $violations,
+        $sortProperties = true,
+        $useAllLanguages = true,
+        array $targetLocales = [],
+        $domain = 'messages'
+    ) {
+        $violationList = iterator_to_array($violations, false);
+        $hasLocales    = count($targetLocales) > 0;
+
+        if (!$useAllLanguages && $hasLocales || $useAllLanguages && !$hasLocales) {
+            throw new \InvalidArgumentException(
+                'Wrong usage of $targetLocales: If all languages should be rendered, $targetLocales should be set, if not, $targetLocales should be an empty array!'
+            );
+        }
+
+        return array_reduce(
+            $violationList,
+            function (array $carry, ConstraintViolation $violation) use (
+                $targetLocales,
+                $domain,
+                $useAllLanguages,
+                $sortProperties
+            ) {
+                // Every translation entry (whether sorted by properties in deeper levels or as top level)
+                // should provide a basic structure.
+                $structure = [];
+                switch (true) {
+                    // If sorted by $targetLanguages, it may look like this:
+                    //
+                    // [
+                    //   'de' => ['Deutscher Text'],
+                    //   'en' => ['English text'],
+                    // ]
+                    //
+                    // When sorting by property, the structure will be merged recursively into the property list:
+                    //
+                    // [
+                    //   'property' => [
+                    //     // this is the structure merged into the property.
+                    //     'de' => ['Deutscher Text'],
+                    //     'en' => ['English text'],
+                    //   ]
+                    // ]
+                    case $useAllLanguages:
+                        foreach ($targetLocales as $locale) {
```

Comment:
another point where we should evaluate whether the imperative approach or a ``array_`` function is the better solution